lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -1055,6 +1055,22 @@ def test_query_counts(self, (sql_calls, mongo_calls, self_paced), self_paced_ena
             resp = views.progress(self.request, course_id=unicode(self.course.id))
         self.assertEqual(resp.status_code, 200)
 
+    @patch('courseware.grades.grade', Mock(return_value={
+        'grade': 'Pass', 'percent': 0.75, 'section_breakdown': [], 'grade_breakdown': []
+    }))
+    def test_generate_cert_for_mode(self):
```

Comment:
You can use [`ddt`](http://ddt.readthedocs.org/en/latest/) to test this for all modes:

```python
@ddt.unpack
@ddt.data(
  (CourseMode.Audit, False),
  (CourseMode.Verified, True),
  ...
)
def test_show_certificate_request_button(self, course_mode, show_button):
  ...
  self.assertEqual(show_button, 'Request Certificate' in response)
```